inc/notificationtarget.class.php

```diff
@@ -184,9 +186,15 @@ function getFromDBForTarget($notifications_id, $type, $ID) {
    function validateSendTo($event, array $infos, $notify_me = false) {
 
       if (!$notify_me) {
-         if (isset($infos['users_id'])
-            // Check login user and not event launch by crontask
-             && ($infos['users_id'] === Session::getLoginUserID(false))) {
+         $last_updater_id = null;
```

Comment:
When the mailcollector work by cron, you don't have a session.

Reply:
Yes, it is exactly why I introduced ability to get the last updater id from the object, and not using the current session.

Reply:
And how do you know that the last updater is the author of the mail?
You can have many people working in the same ticket.

Reply:
I guess "last_updater_id" field corresponds to the last updater of the ticket.
But the problem is not here. Some notifications could be made to warn users about the state of an object, independently from its update (For example warn user that a ticket he creates was not taken into accouint after a defined period). In this case, notification have to be sent to last updater.

So the only valid solution I see for the moment is to change the signature of `NotificationEvent::raiseEvent()`, `NotificationEventInterface::raise()` and `NotificationTarget::validateSendTo()` to add an optionnal parameter `$trigerrer_user_id` that can be used to force identification of the user that triggers the notification.